src/nodes.coffee

```diff
@@ -2883,7 +2896,11 @@ exports.ExecutableClassBody = class ExecutableClassBody extends Base
         # The class scope is not available yet, so return the assignment to update later
         assign = @externalCtor = new Assign new Value, value
       else if not assign.variable.this
-        name      = new (if base.shouldCache() then Index else Access) base
+        name =
+          if base instanceof ComputedPropertyName
+            new Index base.value
+          else
+            new (if base.shouldCache() then Index else Access) base
```

Comment:
If you remove the conditional logic here (so just `new Access base`) no tests break. But I don't feel completely confident removing it. I traced the code's origin back to #4354